core/node/base_token_adjuster/src/base_token_ratio_provider.rs

```diff
@@ -0,0 +1,124 @@
+use std::{fmt::Debug, num::NonZeroU64, time::Duration};
+
+use anyhow::Context;
+use async_trait::async_trait;
+use tokio::sync::watch;
+use zksync_dal::{ConnectionPool, Core, CoreDal};
+use zksync_types::fee_model::BaseTokenConversionRatio;
+
+const CACHE_UPDATE_INTERVAL: Duration = Duration::from_millis(500);
```

Comment:
Does it have to be this small? IIRC we use a single value for the whole batch, so updating it very often may result in unpredictable behavior. I don't have any concrete suggestions at this point, but please thing through the whole flow to make sure that both UX and operator's predictability are good.

Reply:
I have thought through this. There's a funky thing with our fee model that we
- take one value for the entire batch
- upon estimateGas, take the higher of (prev block and current estimation)
So half a second, to me, captures the tradeoff of reflecting current behavior while not caching excessively. The only thing it doesn't capture is that in reality, operators are unlikely to fetch new ratios from external APIs every half a second. We will keep this value in mind as we observe patterns of external fetching frequency

Reply:
Up to you. You may even make it an optional configuration parameter if you want (as you did [here](https://github.com/matter-labs/zksync-era/blob/cfdeb10f0fa8518e74790ec7099923f7546fa2a4/core/lib/config/src/configs/base_token_adjuster.rs#L10)).

Reply:
Oh yes, maybe that's the way to go indeed! I added an explicit ticket to tackle this as a small add once we have more info.